src/linear_algebra/direct_sum/finsupp.lean

```diff
@@ -31,7 +31,7 @@ variables (R M) (ι : Type*) [decidable_eq ι]
 copies of M indexed by ι. -/
 def finsupp_lequiv_direct_sum : (ι →₀ M) ≃ₗ[R] direct_sum ι (λ i, M) :=
 linear_equiv.of_linear
-  (finsupp.lsum $ direct_sum.lof R ι (λ _, M))
+  (finsupp.lsum $ (show ι → M →ₗ[R] direct_sum ι (λ i, M), from direct_sum.lof R ι (λ _, M)))
```

Comment:
Is this better or worse as an ascription?
```suggestion
  (finsupp.lsum $ (direct_sum.lof R ι (λ _, M) : ι → M →ₗ[R] direct_sum ι (λ i, M)))
```

I've seen comments about `show` introducing extra levels of unwrapping that type ascriptions do not

Reply:
Does it work? I tried to use type ascription here and failed.

Reply:
I haven't tried. This looks like something I ran into recently, so maybe I'll check out the branch and have a go myself.

Reply:
You're right, I can't get anywhere here either. You can make this a bit shorter with `open_locale direct_sum` and using `(⨁ i, M)` instead of `direct_sum ι (λ i, M)`, but that's fine to declare out of scope.